src/Emulator/Peripherals/Peripherals/Timers/STM32_Timer.cs

```diff
@@ -329,12 +293,31 @@ public STM32_Timer(Machine machine, long frequency, uint initialLimit) : base(ma
 
         public uint ReadDoubleWord(long offset)
         {
-            return registers.Read(offset);
+            switch((Registers)offset)
+            {
+                case Registers.CaptureOrCompareMode1:
+                   return captureOrCompareMode1;
+                case Registers.CaptureOrCompareMode2:
+                   return captureOrCompareMode2;
+                default:
+                    return registers.Read(offset);
+            }            
         }
 
         public void WriteDoubleWord(long offset, uint value)
         {
-            registers.Write(offset, value);
+            switch((Registers)offset)
+            {
+                case Registers.CaptureOrCompareMode1:
```

Comment:
this should effectively be implemented as relevant flags/fields in relevant registers. Otherwise, again, we just end up with and r/w field with no logic